### ISSUE TITLE
Failed to install AWS-SAM-CLI on a Amazon Ubuntu instance

### ISSUE THREAD

**Issue by smpavlenko** [aws/aws-sam-cli#2260]
### Description
Failure to install AWS-SAM-CLI on a Amazon Ubuntu instance (us-east-1).

```
> cat /etc/os-release

NAME="Ubuntu"
VERSION="16.04.7 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.7 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

### Steps to reproduce
Followed https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install-linux.html

1. Installed Homebrew
```
> brew --version

Homebrew 2.5.2
Homebrew/linuxbrew-core (git revision c5f9d1; last commit 2020-09-29)
```

2. Installed AWS SAM CLI
```
> brew install aws-sam-cli

==> Installing aws-sam-cli from aws/tap
==> Downloading https://linuxbrew.bintray.com/bottles/patchelf-0.11.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/73/7396e6cf5dbe62fae0e7fcba57f88586359561824df9cfe1127b4d88dd6f2a5a?__gda__=exp=1601459272~hmac=f492e9844c623f
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/gdbm-1.18.1_1.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/dd/dd00a26fa20413f81477af032587de19bb620eef352a6d8dd3d9c3a176f6bd5a?__gda__=exp=1601459272~hmac=f70ee7f992646c
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/openssl%401.1-1.1.1g.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/64/6487415cb2902e5837927427a6b93a579bf5481652e526422d73f687bef41d3e?__gda__=exp=1601459273~hmac=bc0ef596699bd5
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/ncurses-6.2.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/fc/fc5a0983a8afae8ef46945f3bec9dba9d66e81e879027d100d05088ea6f8043d?__gda__=exp=1601459274~hmac=5ab025a939a670
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/readline-8.0.4.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/46/46da47db3da04b2f248e3cf2d6d14c55aa543555d1134f6cbbf07787a5bf0bd6?__gda__=exp=1601459275~hmac=705c6ae9d8b95b
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/zlib-1.2.11.x86_64_linux.bottle.tar.gz
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/sqlite-3.33.0.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/ba/bafd40d3bcbfe8350a04bd65fd2fced3ea98a74b27c6102972eca055d1eca8fe?__gda__=exp=1601459277~hmac=71b9befd2cca1c
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/xz-5.2.5.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/67/676b0e178b3e7644f86385cb2497ac5ec490e2222ba3d0147e28bd85aff365cf?__gda__=exp=1601459278~hmac=7425f56f027286
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/bzip2-1.0.8.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/e5/e5fce257b9fee119d28d0e7f7dac9332bd26b248fe35d983ba85104243e4bc2e?__gda__=exp=1601459279~hmac=32e8c4a43946fd
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/libffi-3.3.x86_64_linux.bottle.tar.gz
######################################################################## 100.0%
==> Downloading https://linuxbrew.bintray.com/bottles/python%403.7-3.7.9.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/bb/bb1f3e3546faf9ced7dfe408049be9e46a270efca859cb3819f0894fcccd3e18?__gda__=exp=1601459281~hmac=1f8d55b28511d1
######################################################################## 100.0%
==> Downloading https://github.com/aws/aws-sam-cli/releases/download/v1.3.2//aws-sam-cli-1.3.2.x86_64_linux.bottle.tar.gz
==> Downloading from https://github-production-release-asset-2e65be.s3.amazonaws.com/92205085/4c24c000-fe7f-11ea-82fb-3e694664e907?X-Amz-Algorithm=AWS4-HMAC-S
######################################################################## 100.0%
==> Installing dependencies for aws/tap/aws-sam-cli: patchelf, gdbm, openssl@1.1, ncurses, readline, zlib, sqlite, xz, bzip2, libffi and python@3.7
==> Installing aws/tap/aws-sam-cli dependency: patchelf
==> Pouring patchelf-0.11.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/patchelf/0.11: 7 files, 909.2KB
==> Installing aws/tap/aws-sam-cli dependency: gdbm
==> Pouring gdbm-1.18.1_1.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/gdbm/1.18.1_1: 40 files, 1MB
==> Installing aws/tap/aws-sam-cli dependency: openssl@1.1
==> Pouring openssl@1.1-1.1.1g.x86_64_linux.bottle.tar.gz
==> Downloading https://curl.haxx.se/ca/cacert-2020-01-01.pem
######################################################################## 100.0%
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /home/linuxbrew/.linuxbrew/etc/openssl@1.1/certs

and run
  /home/linuxbrew/.linuxbrew/opt/openssl@1.1/bin/c_rehash
==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/openssl@1.1/1.1.1g: 8,060 files, 21.4MB
==> Installing aws/tap/aws-sam-cli dependency: ncurses
==> Pouring ncurses-6.2.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/ncurses/6.2: 3,901 files, 9.5MB
==> Installing aws/tap/aws-sam-cli dependency: readline
==> Pouring readline-8.0.4.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/readline/8.0.4: 53 files, 1.9MB
==> Installing aws/tap/aws-sam-cli dependency: zlib
==> Pouring zlib-1.2.11.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11: 12 files, 424KB
==> Installing aws/tap/aws-sam-cli dependency: sqlite
==> Pouring sqlite-3.33.0.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/sqlite/3.33.0: 12 files, 4.9MB
==> Installing aws/tap/aws-sam-cli dependency: xz
==> Pouring xz-5.2.5.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/xz/5.2.5: 128 files, 1.8MB
==> Installing aws/tap/aws-sam-cli dependency: bzip2
==> Pouring bzip2-1.0.8.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/bzip2/1.0.8: 30 files, 549.3KB
==> Installing aws/tap/aws-sam-cli dependency: libffi
==> Pouring libffi-3.3.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/libffi/3.3: 17 files, 552.7KB
==> Installing aws/tap/aws-sam-cli dependency: python@3.7
==> Pouring python@3.7-3.7.9.x86_64_linux.bottle.tar.gz
==> /home/linuxbrew/.linuxbrew/Cellar/python@3.7/3.7.9/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/home/linuxbrew/.linu
==> /home/linuxbrew/.linuxbrew/Cellar/python@3.7/3.7.9/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/home/linuxbrew/.linu
==> /home/linuxbrew/.linuxbrew/Cellar/python@3.7/3.7.9/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/home/linuxbrew/.linu
==> Caveats
Python has been installed as
  /home/linuxbrew/.linuxbrew/opt/python@3.7/bin/python3

Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /home/linuxbrew/.linuxbrew/opt/python@3.7/libexec/bin

You can install Python packages with
  /home/linuxbrew/.linuxbrew/opt/python@3.7/bin/pip3 install <package>
They will install into the site-package directory
  /home/linuxbrew/.linuxbrew/lib/python3.7/site-packages

See: https://docs.brew.sh/Homebrew-and-Python

python@3.7 is keg-only, which means it was not symlinked into /home/linuxbrew/.linuxbrew,
because this is an alternate version of another formula.

If you need to have python@3.7 first in your PATH run:
  echo 'export PATH="/home/linuxbrew/.linuxbrew/opt/python@3.7/bin:$PATH"' >> ~/.profile

For compilers to find python@3.7 you may need to set:
  export LDFLAGS="-L/home/linuxbrew/.linuxbrew/opt/python@3.7/lib"
  export CPPFLAGS="-I/home/linuxbrew/.linuxbrew/opt/python@3.7/include"

==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/python@3.7/3.7.9: 3,656 files, 67.2MB
==> Installing aws/tap/aws-sam-cli
==> Pouring aws-sam-cli-1.3.2.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/aws-sam-cli/1.3.2: 4,147 files, 90.8MB
==> No outdated dependents to upgrade!
==> Checking for dependents of upgraded formulae...
==> Reinstalling 1 broken dependent from source:
aws/tap/aws-sam-cli
==> Caveats
==> openssl@1.1
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /home/linuxbrew/.linuxbrew/etc/openssl@1.1/certs

and run
  /home/linuxbrew/.linuxbrew/opt/openssl@1.1/bin/c_rehash
==> python@3.7
Python has been installed as
  /home/linuxbrew/.linuxbrew/opt/python@3.7/bin/python3

Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /home/linuxbrew/.linuxbrew/opt/python@3.7/libexec/bin

You can install Python packages with
  /home/linuxbrew/.linuxbrew/opt/python@3.7/bin/pip3 install <package>
They will install into the site-package directory
  /home/linuxbrew/.linuxbrew/lib/python3.7/site-packages

See: https://docs.brew.sh/Homebrew-and-Python

python@3.7 is keg-only, which means it was not symlinked into /home/linuxbrew/.linuxbrew,
because this is an alternate version of another formula.

If you need to have python@3.7 first in your PATH run:
  echo 'export PATH="/home/linuxbrew/.linuxbrew/opt/python@3.7/bin:$PATH"' >> ~/.profile

For compilers to find python@3.7 you may need to set:
  export LDFLAGS="-L/home/linuxbrew/.linuxbrew/opt/python@3.7/lib"
  export CPPFLAGS="-I/home/linuxbrew/.linuxbrew/opt/python@3.7/include"
```

3. Check version
```
> sam --version

The program 'sam' is currently not installed. To run 'sam' please ask your administrator to install the package 'simon'
```

4. Install SAM CLI second time
```
> brew install aws-sam-cli

==> Installing aws-sam-cli from aws/tap
==> Downloading https://github.com/aws/aws-sam-cli/releases/download/v1.3.2//aws-sam-cli-1.3.2.x86_64_linux.bottle.tar.gz
Already downloaded: /home/dev/.cache/Homebrew/downloads/08da90c30637ee3f5cb0f6a1da606d77c5178f40f6b72304a286bb3ee3567742--aws-sam-cli-1.3.2.x86_64_linux.bottle.tar.gz
==> Pouring aws-sam-cli-1.3.2.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/aws-sam-cli/1.3.2: 4,147 files, 90.8MB
```

5. Check version
```
> sam --version

SAM CLI, version 1.3.2
```

### Expected result
AWS SAM CLI installed after first installation

**Comment by major-mayer** [comment]
Same problem on my end. Thanks for the hint, installing it once again did the trick :+1:

**Comment by fbau3r** [comment]
We seem to have the same problem (and executing `brew install aws-sam-cli` twice does work for us too).

From additional trial and error we found that executing `brew install python@3.8` (a dependency of `aws-sam-cli`) before executing `brew install aws-sam-cli` worked. The lines `Reinstalling 1 broken dependent from source:` and `aws/tap/aws-sam-cli` seem to be the point where the issue takes place (the reinstall directory in the Cellar is pretty empty).

I don't know if it is of any help, but since we've encountered the issue within a docker container, I attached the Dockerfiles and a README how we reproduced the problem (`Dockerfile-Homebrew2a`) and how we solved the problem with pre-installing the `python@3.8`-dependency (`Dockerfile-Homebrew2b`):

- [Homebrew-aws-sam-cli.zip](https://github.com/aws/aws-sam-cli/files/5523685/Homebrew-aws-sam-cli.zip)  
    Maybe this helps with the investigation...

```
> brew --version

Homebrew 2.5.9
Homebrew/linuxbrew-core (git revision 9b37ceb; last commit 2020-11-11)
```

`aws-sam-cli` was this bottle at the moment of our installation: `https://github.com/aws/aws-sam-cli/releases/download/v1.9.0//aws-sam-cli-1.9.0.x86_64_linux.bottle.tar.gz`.
